cmd/system-shutdown/system-shutdown.c

```diff
@@ -56,6 +62,8 @@ int main(int argc, char *argv[])
 	if (mkdir("/writable", 0755) < 0) {
 		die("cannot create directory /writable");
 	}
+	// We are reading a file from /run and need to do this before unmounting
+	sc_read_reboot_arg(reboot_arg, sizeof reboot_arg);
```

Comment:
Yes, because no action would be taken if there is a failure to read the file, we will simply have reboot_arg untouched, which is checked later when choosing how to reboot. If you want we can print something here, but just an info log, as the file not being present would be actually quite frequent.